### DISCUSSION TITLE
ctx update causes infinite loop

### DISCUSSION THREAD

**Discussion by wchen** [evrete/evrete#93] [General]
I have a simple test program attached. (need rename to java file since I can not attach java files directly)
[DemoMain.txt](https://github.com/evrete/evrete/files/14971215/DemoMain.txt)
[TestObject.txt](https://github.com/evrete/evrete/files/14971216/TestObject.txt)

The DemoMain will get into infinite loop if I uncomment ctx.update to the rule.

Any idea?

**Comment by andbi** [comment]
This is expected behavior. By updating a fact, you're telling the engine to re-evaluate conditions, and that condition still holds after the update. However, if you changed the `trigger` value, there would be no infinite loop.
